docs/logs.md

```diff
@@ -34,3 +34,69 @@ You can get execution logs using one of the following methods:
 - Get the logs using [Tekton Dashboard](https://github.com/tektoncd/dashboard).
 
 - Configure an external service to consume and display the logs. For example, [ElasticSearch, Beats, and Kibana](https://github.com/mgreau/tekton-pipelines-elastic-tutorials).
+
+
+# Logs Credential Filter
+
+Often it is the case that secret values will sneak into the ouput logs of your pipelines. You can either handle this manually by avoiding to print them in the output or use the feature for filtering secret values from the output log. To enable this feature set the `enable-logging-credentials-filter` value in the `feature-flags` config map to `"true"` (see [Customizing the Pipelines Controller behavior](install.md#customizing-the-pipelines-controller-behavior)).
+
+The tekton controller looks for secret refs in volumes and env and also for CSI volumes referencing secrets in the definition of the pipeline pod. It saves the information in form of the names of environment variables and paths to files with secrets. The pod can then read those values and redact them.
+
+This design does not transmit actual secrets and the pod does not need access to the API sever to retrieve them. It tells the pipeline pod only the places where it can find secrets attached to it. And those are the secrets that will be redacted. Because only those can be used in the corresponding pipeline step.
+
+The credential filter will redact values contained in those secrets from the output log stream and replace them with `[REDACTED:<secret-name>]`.
+
+## Secret Detection Logic
```

Comment:
Could you also provide an example TaskRun with a secret mounted as a workspace that has the secret scrubbed?

Reply:
What exactly do you mean with "secret mounted as a workspace"?

Reply:
Well, I did not know that feature. 
I will have a look into it and extend the implementation.

Reply:
I extended the code to also test the integration with workspaces. In the end its the same working code as before I only added some more test cases.

I also discovered that I did not yet cover projected volumes with secrets and added that to the logic too.